test/readremote.jl

```diff
@@ -3,186 +3,183 @@ using Base.Test
 
 @testset "readremote" begin
 
-    # check that we can read the official HDF5 example files
-    urlbase = "https://support.hdfgroup.org/ftp/HDF5/examples/files/exbyapi/"
-
-    fcmp = [0 1 2 3 4 5 6;
-        2 1.66667 2.4 3.28571 4.22222 5.18182 6.15385;
-        4 2.33333 2.8 3.57143 4.44444 5.36364 6.30769;
-        6 3 3.2 3.85714 4.66667 5.54545 6.46154]'
-    icmp = [0 -1 -2 -3 -4 -5 -6;
-        0 0 0 0 0 0 0;
-        0 1 2 3 4 5 6;
-        0 2 4 6 8 10 12]'
-    const SOLID, LIQUID, GAS, PLASMA = 0, 1, 2, 3
-    ecmp = [SOLID SOLID SOLID SOLID SOLID SOLID SOLID;
-            SOLID LIQUID GAS PLASMA SOLID LIQUID GAS;
-            SOLID GAS SOLID GAS SOLID GAS SOLID;
-            SOLID PLASMA GAS LIQUID SOLID PLASMA GAS]'
-    scmp = ["Parting", "is such", "sweet", "sorrow."]
-    vicmp = Array{Int32}[[3,2,1],[1,1,2,3,5,8,13,21,34,55,89,144]]
-    opq = Array{UInt8}[[0x4f, 0x50, 0x41, 0x51, 0x55, 0x45, 0x30],
-                       [0x4f, 0x50, 0x41, 0x51, 0x55, 0x45, 0x31],
-                       [0x4f, 0x50, 0x41, 0x51, 0x55, 0x45, 0x32],
-                       [0x4f, 0x50, 0x41, 0x51, 0x55, 0x45, 0x33]]
-    # For H5T_ARRAY
-    AA = Array{Int,2}[
-    [ 0   0   0;
+# check that we can read the official HDF5 example files
+urlbase = "https://support.hdfgroup.org/ftp/HDF5/examples/files/exbyapi/"
+
+fcmp = [0 1 2 3 4 5 6;
+    2 1.66667 2.4 3.28571 4.22222 5.18182 6.15385;
+    4 2.33333 2.8 3.57143 4.44444 5.36364 6.30769;
+    6 3 3.2 3.85714 4.66667 5.54545 6.46154]'
+icmp = [0 -1 -2 -3 -4 -5 -6;
+    0 0 0 0 0 0 0;
+    0 1 2 3 4 5 6;
+    0 2 4 6 8 10 12]'
+const SOLID, LIQUID, GAS, PLASMA = 0, 1, 2, 3
+ecmp = [SOLID SOLID SOLID SOLID SOLID SOLID SOLID;
+        SOLID LIQUID GAS PLASMA SOLID LIQUID GAS;
+        SOLID GAS SOLID GAS SOLID GAS SOLID;
+        SOLID PLASMA GAS LIQUID SOLID PLASMA GAS]'
+scmp = ["Parting", "is such", "sweet", "sorrow."]
+vicmp = Array{Int32}[[3,2,1],[1,1,2,3,5,8,13,21,34,55,89,144]]
+opq = Array{UInt8}[[0x4f, 0x50, 0x41, 0x51, 0x55, 0x45, 0x30],
+                   [0x4f, 0x50, 0x41, 0x51, 0x55, 0x45, 0x31],
+                   [0x4f, 0x50, 0x41, 0x51, 0x55, 0x45, 0x32],
+                   [0x4f, 0x50, 0x41, 0x51, 0x55, 0x45, 0x33]]
+# For H5T_ARRAY
+AA = Array{Int,2}[
+    [0   0   0;
      0  -1  -2;
      0  -2  -4;
      0  -3  -6;
      0  -4  -8],
-    [ 0  1   2;
+    [0  1   2;
      1  1   1;
      2  1   0;
      3  1  -1;
      4  1  -2],
-    [ 0  2  4;
+    [0  2  4;
      2  3  4;
      4  4  4;
      6  5  4;
      8  6  4],
-    [ 0   3   6;
-      3   5   7;
-      6   7   8;
-      9   9   9;
+    [0   3   6;
+     3   5   7;
+     6   7   8;
+     9   9   9;
      12  11  10]]
 
 
-    const savedir = joinpath(tempdir(), "h5")
-    if !isdir(savedir)
-        mkdir(savedir)
+const savedir = mktempdir()
```

Comment:
Since these are downloading specific known files instead of creating some throwaway local file, it may be better to save them somewhere consistent. Giving it a unique name will result in repeated downloads, a predictable name will only download them once.